### PR TITLE
Adding directory for joint imaging scripts

### PR DESCRIPTION
**Pull request by d-l-walker**

### CODE REVIEW

**Comment by keflavich** [comment]
You don't need a copy of `default_tclean_commands.json` since that already exists elsewhere in the repository; I'd recommend removing that from this PR.

File: joint_deconvolution/joint_deconvolution_and_feather_auto.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+with open(f"./default_tclean_commands.json", "r") as fh:
+    tclean_commands = json.load(fh)
+
+os.chdir(pth+'/'+region)
+if not os.path.exists('clean_12m_7m'):
+    os.mkdir('clean_12m_7m')
+os.chdir(pth+'/'+region+'/clean_12m_7m/')
+
+vis_all     = glob.glob(pth+'/'+region+'/twelve_m/spws/*.spw3') + glob.glob(pth+'/'+region+'/seven_m/spws/*.spw3')
```
Comment (keflavich) [inline joint_deconvolution/joint_deconvolution_and_feather_auto.py:14]: general recommendation (python-wide): either do
```suggestion
vis_all     = glob.glob(os.path.join(pth, region, '/twelve_m/spws/*.spw3')) + glob.glob(os.path.join(region, '/seven_m/spws/*.spw3'))
```
or
`f'{pth}/{region}/twelve/...`

The former is more robust (it works on windows, for example), while the latter is easier to read

File: joint_deconvolution/joint_deconvolution_and_feather_auto.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import json, os, sys, glob
+
+region      = sys.argv[1]
+pth         = os.path.abspath(os.getcwd())
+
+with open(f"./default_tclean_commands.json", "r") as fh:
```
Comment (keflavich) [inline joint_deconvolution/joint_deconvolution_and_feather_auto.py:6]: if we don't have `default_tclean_commands.json` in the same directory, as I'm suggesting above, we should do something like what we do [elsewhere](https://github.com/ACES-CMZ/reduction_ACES/blob/158e54b4b2f9ca0900832e62755c4981cf977668/retrieval_scripts/run_pipeline.py#L45), i.e.:
```python
rootdir = os.getenv['ACES_ROOTDIR']
with open(f"{rootdir}/pipeline_scripts/default_tclean_commands.json", "r") as fh:
```

File: joint_deconvolution/joint_deconvolution_and_feather_auto.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    tclean_commands = json.load(fh)
+
+os.chdir(pth+'/'+region)
+if not os.path.exists('clean_12m_7m'):
+    os.mkdir('clean_12m_7m')
+os.chdir(pth+'/'+region+'/clean_12m_7m/')
+
+vis_all     = glob.glob(pth+'/'+region+'/twelve_m/spws/*.spw3') + glob.glob(pth+'/'+region+'/seven_m/spws/*.spw3')
+if os.path.isdir(region+'_12m_7m_SPW3_concat.ms')==False:
```
Comment (keflavich) [inline joint_deconvolution/joint_deconvolution_and_feather_auto.py:15]:
```suggestion
if not os.path.isdir(region+'_12m_7m_SPW3_concat.ms'):
```

`==False` is discouraged

File: joint_deconvolution/joint_deconvolution_and_feather_auto.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    concat(vis       = vis_all,
+           concatvis = region+'_12m_7m_SPW3_concat.ms')
+
+hnco_pars   = tclean_commands[region+'_TM1']['tclean_cube_pars']['spw31']
+
+if os.path.isdir(pth+'/'+region+'/clean_12m_7m/'+region+'_12m_7m_SPW3.clean.image')==False and os.path.isdir(pth+'/'+region+'/clean_12m_7m/'+region+'_12m_7m_TP.SPW3')==False:
```
Comment (keflavich) [inline joint_deconvolution/joint_deconvolution_and_feather_auto.py:21]:
```suggestion
if not os.path.isdir(pth+'/'+region+'/clean_12m_7m/'+region+'_12m_7m_SPW3.clean.image') and not os.path.isdir(pth+'/'+region+'/clean_12m_7m/'+region+'_12m_7m_TP.SPW3'):
```

File: joint_deconvolution/joint_deconvolution_and_feather_auto.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+           noisethreshold          = hnco_pars['noisethreshold'],
+           lownoisethreshold       = hnco_pars['lownoisethreshold'],
+           negativethreshold       = hnco_pars['negativethreshold'],
+           minbeamfrac             = hnco_pars['minbeamfrac'],
+           growiterations          = hnco_pars['growiterations'],
+           dogrowprune             = hnco_pars['dogrowprune'])
+
+tp_data = glob.glob(pth+'/'+region+'/total_power/'+'*.spw23.cube.I.sd.fits')[0]
+
+if os.path.isdir(pth+'/'+region+'/clean_12m_7m/'+region+'_12m_7m_SPW3.clean.image')==True:
```
Comment (keflavich) [inline joint_deconvolution/joint_deconvolution_and_feather_auto.py:62]:
```suggestion
if os.path.isdir(pth+'/'+region+'/clean_12m_7m/'+region+'_12m_7m_SPW3.clean.image'):
```

File: joint_deconvolution/joint_deconvolution_and_feather_auto.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    tp_freq       = imhead(tp_data,mode='get',hdkey='restfreq')
+    high_res_freq = imhead(pth+'/'+region+'/clean_12m_7m/'+region+'_12m_7m_SPW3.clean.image',mode='get',hdkey='restfreq')
+
+    if tp_freq['value'] != high_res_freq['value']:
+        imreframe(imagename = tp_data,
+                  restfreq  = high_res_freq['value']+' Hz',
+                  output    = tp_data+'.reframe')
+
+    if os.path.isdir(tp_data+'.reframe')==True:
```
Comment (keflavich) [inline joint_deconvolution/joint_deconvolution_and_feather_auto.py:71]:
```suggestion
    if os.path.isdir(tp_data+'.reframe'):
```

**Review by keflavich — CHANGES_REQUESTED** [review]
Don't track `.pyc` files, they are junk.

I suggest adopting some different python patters generally.  I gave a few examples.  These are mostly minor - no functionality change - but it's good practice to adopt "pythonic" norms.

I suggest changing path handling, though - there are some awkward hard-coded path assumptions that we should make more modular.

**Comment by d-l-walker** [comment]
Thanks Adam! 
Yeah, I was aware that the hard-coded path assumptions were not ideal. I should have had this in mind during my initial testing, but ended up just going with a local structure. When I get the time, I will try to restructure this so that it works in the general ACES structure.

And yeah, I forgot to remove the unnecessary files (.pyc, .json) before pushing this. Thanks for catching that.

**Comment by keflavich** [comment]
Sounds good.   We can merge any time you want just to have the base code in the repository, or we can leave this PR open until you're ready to refactor the paths.

**Comment by d-l-walker** [comment]
I'm happy for this to be merged as-is. Once it's in the main repo I'll just work from there when I get a chance to sort the paths out. I see that merging is blocked -- I think maybe you need to review the changes that were made based on your suggestions? If I'm wrong and I can do that myself, let me know!

**Review by keflavich — APPROVED** [review]

**Pull request merged**